ring/ring.go

```diff
@@ -334,6 +320,7 @@ func (r *Ring) updateRingState(ringDesc *Desc) {
 		// when watching the ring for updates).
 		r.mtx.Lock()
 		r.ringDesc = ringDesc
+		r.updateRingMetrics(rc)
```

Comment:
[nit] Given this is the common case and given how easy it is to move it to read lock only, we could take the read lock only for this case.

Reply:
We discussed the locking IRL and opted to keep the locking outside updateRingMetrics